transformers/uws/server.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+var parse = require('url').parse
+  , http = require('http');
+
+/**
+ * Server of µWebSockets transformer.
+ *
+ * @runat server
+ * @api private
+ */
+module.exports = function server() {
+  var uws = require('uws').uws
+    , Spark = this.Spark;
+
+  var service = this.service = new uws.Server();
+
+  service.onMessage(function message(socket, msg, binary) {
+    var spark = service.getData(socket);
+    spark.emit('incoming::data', binary ? msg : msg.toString());
+  });
+
+  service.onDisconnection(function close(socket) {
+    var spark = service.getData(socket);
+    service.setData(socket);
+    spark.emit('incoming::end');
+  });
+
+  //
+  // Listen to upgrade requests.
+  //
+  this.on('upgrade', function upgrade(req, soc, head) {
+    this.service.onConnection(function create(socket) {
```

Comment:
@3rd-Eden do you think it will be better to go with something like this https://github.com/alexhultman/uWebSockets/issues/45#issuecomment-208740046?